packages/web/integrations/nomic/transfer.tsx

```diff
@@ -0,0 +1,502 @@
+    const [proceeded, setProceeded] = useState(isWithdraw);
+    const [reachedCapacityLimit, setReachedCapacityLimit] = useState<
+      boolean | undefined
+    >(undefined);
+    const [qrBlob, setQrBlob] = useState<string | undefined>(undefined);
+    const [depositAddress, setDepositAddress] = useState<string | undefined>(
+      undefined
+    );
+    const [pendingDepositAmount, setPendingDepositAmount] = useState<
+      number | undefined
+    >(undefined);
+    const [bridgeFee, setBridgeFee] = useState<number | undefined>(undefined);
+    const [minerFee, setMinerFee] = useState<number | undefined>(undefined);
+    const [withdrawAmount, setWithdrawAmount] = useState("");
+    const [withdrawAddress, setWithdrawAddress] = useState("");
+
+    const nomicChainId = IS_TESTNET ? "nomic-testnet-4d" : "nomic-stakenet-3";
+
+    useEffect(() => {
+      if (!osmosisAccount || !osmosisAccount.address) return;
+      const relayers = IS_TESTNET ? ["http://localhost:8999"] : [];
+      generateDepositAddress({
+        relayers,
+        channel: IS_TESTNET ? "channel-0" : "",
+        network: IS_TESTNET ? "testnet" : "bitcoin",
+        receiver: osmosisAccount.address,
+      }).then((res) => {
+        if (res.code === 0) {
+          setBridgeFee(res.bridgeFeeRate);
+          setMinerFee(res.minerFeeRate);
+          setDepositAddress(res.bitcoinAddress);
+          setQrBlob(res.qrCodeData);
+          setReachedCapacityLimit(false);
+        } else {
+          if (res.code === 2) {
+            setReachedCapacityLimit(true);
+            return;
+          }
+
+          displayToast(
+            {
+              message: "Unknown Error",
+              caption: res.reason,
+            },
+            ToastType.ERROR
+          );
+        }
+      });
+
+      getPendingDeposits(relayers, osmosisAccount.address).then((deposits) => {
+        setPendingDepositAmount(
+          deposits.reduce((acc: number, deposit: DepositInfo) => {
+            acc += deposit.amount;
+            return acc;
+          }, 0)
+        );
+      });
+    }, [osmosisAccount, isWithdraw]);
+
+    const feeConfig = useFakeFeeConfig(
+      chainStore,
+      chainId,
+      osmosisAccount?.cosmos.msgOpts.ibcTransfer.gas ?? 0
+    );
+    const withdrawAmountConfig = useAmountConfig(
+      chainStore,
+      queriesStore,
+      chainId,
+      address,
+      feeConfig,
+      balanceOnOsmosis.balance.currency
+    );
+
+    async function withdraw() {
+      let memo = `withdraw:${withdrawAddress}`;
+      if (!osmosisAccount || !osmosisAccount.address) {
+        displayToast(
+          {
+            message: "Osmosis Account Error",
+            caption: "Osmosis account not found",
+          },
+          ToastType.ERROR
+        );
+        return;
+      }
+      await osmosisAccount.cosmos.sendIBCTransferMsg(
+        {
+          portId: "transfer",
+          channelId: IS_TESTNET ? "channel-4109" : "",
```

Comment:
I believe this channel id should come from our chain config. I believe it should be available within `balanceOnOsmosis.sourceChannelId`